### PR TITLE
[numpy] FFI binary bitwise ops 

### PR DESCRIPTION
**Pull request by Yiyan66**
## Description ##
FFI bitwise_and, bitwise_or, bitwise_xor

&#160;  | Old FFI (ctypes) (us) | New FFI (cython) (us)
:-------: | :-------------------: | :-: 
bitwise_and       | 20.96                | 7.34
bitwise_or       | 21.05                 | 7.21
bitwise_xor       | 21.1                 | 7.34
copysign       | 26.08                | 7.32
arctan2       | 26.28                | 7.16
hypot       | 25.93                 | 7.11
ldexp       | 26.56                | 7.24

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/ndarray/numpy/_op.py [hunk 1]
```diff
@@ -5848,7 +5848,9 @@ def bitwise_and(x1, x2, out=None, **kwargs):
     >>> np.bitwise_and(np.array([True, True], dtype='bool'), np.array([False, True], dtype='bool'))
     array([False,  True])
     """
-    return _ufunc_helper(x1, x2, _npi.bitwise_and, _np.bitwise_and, _npi.bitwise_and_scalar, None, out)
+    if isinstance(x1, numeric_types) and isinstance(x2, numeric_types):
+        _np.bitwise_and(x1, x2, out=out)
```
Comment (hzfan) [inline python/mxnet/ndarray/numpy/_op.py]: `return _np.bitwise_and(x1, x2, out=out)`. Same for two other ops.

File: src/api/operator/numpy/np_elemwise_broadcast_op.cc [hunk 2]
```diff
@@ -36,4 +36,29 @@ MXNET_REGISTER_API("_npi.add")
   UFuncHelper(args, ret, op, op_scalar, nullptr);
 });
 
+
```
Comment (haojin2) [inline src/api/operator/numpy/np_elemwise_broadcast_op.cc]: get rid of this blank line.

File: src/api/operator/numpy/np_elemwise_broadcast_op.cc [hunk 3]
```diff
@@ -36,4 +36,60 @@ MXNET_REGISTER_API("_npi.add")
...
+  const nnvm::Op* op_scalar = Op::Get("_npi_copysign_scalar");
+  UFuncHelper(args, ret, op, op_scalar, nullptr);
+});
+
+MXNET_REGISTER_API("_npi.arctan2")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_arctan2");
+  const nnvm::Op* op_scalar = Op::Get("_npi_arctan2_scalar");
+  UFuncHelper(args, ret, op, op_scalar, nullptr);
```
Comment (hzfan) [inline src/api/operator/numpy/np_elemwise_broadcast_op.cc]: arctan2 is not commutative. The last arg should not be nullptr. An example
```
const nnvm::Op* op_rscalar = Op::Get("_npi_rarctan2_scalar");
UFuncHelper(args, ret, op, op_scalar, op_rscalar);
```

File: src/api/operator/numpy/np_elemwise_broadcast_op.cc [hunk 4]
```diff
@@ -36,4 +36,60 @@ MXNET_REGISTER_API("_npi.add")
...
+  const nnvm::Op* op_scalar = Op::Get("_npi_hypot_scalar");
+  UFuncHelper(args, ret, op, op_scalar, nullptr);
+});
+
+MXNET_REGISTER_API("_npi.ldexp")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_ldexp");
+  const nnvm::Op* op_scalar = Op::Get("_npi_ldexp_scalar");
+  UFuncHelper(args, ret, op, op_scalar, nullptr);
```
Comment (hzfan) [inline src/api/operator/numpy/np_elemwise_broadcast_op.cc]: Same

**Pull request merged**